cmake/print_config_summary.cmake

```diff
@@ -0,0 +1,58 @@
+# ############################################################################
+# Copyright (c) 2022 One Identity
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+# ############################################################################
+
+# Lists actual module definition results
+# Behaves similar to the cmake -L option but filters out the relevant information and adds some highlighting
+#
+function(print_config_summary)
+  get_cmake_property(_variableNames VARIABLES)
+  list(SORT _variableNames)
+
+  message(STATUS "-----------------------------")
+  message(STATUS "syslog-ng Open Source Edition ${PKG_CONFIG_PACKAGE_VERSION} configured")
```

Comment:
One last minor thing:

Can we use SYSLOG_NG_VERSION here? pkg-config is a helper tool for compilation, it does not really make sense using one of its variables here ([PACKAGE_VERSION](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Initializing-configure.html) in autotools is actually the project's version).

Reply:
upps, searched for the autotools [PACKAGE_VERSION](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Initializing-configure.html) and when I found a reference in the cmake files I blindly used the first one :S , shame on me

Reply:
Thank you.